### ISSUE TITLE
3D particle interpolation test

### ISSUE THREAD

**Issue by sac-bsa** [geodynamics/aspect#3208]
I have been working on making the bi-linear least squares particle interpolator work in for three dimensions. I was looking for some help to make a 3D particle example, as I didn't find any existing cookbooks, benchmarks, or tests that used particles and were in 3D. @egpuckett suggested that we use a falling ball in Stokes Flow, known as a Stokes Sphere. I saw that there is a "Stokes' law" benchmark/cookbook already made for 3D. Would this existing parameter file be a good candidate for modification for use as a test for 3D particle interpolation, or is there a different situation that would better test the correctness of particle interpolation in 3D?

**Comment by tjhei** [comment]
I would take simpler 2d particle tests and make them 3d, for example:
- particle_linear_accelerated_flow_rk2.prm is trivial to do in 3d
- You can probably adjust the van keken test to 3d as well, used in many tests (particle_integrator_euler.prm for example)
- prescribed_stokes_solution.prm is a static rotation that should work in 3d

**Comment by gassmoeller** [comment]
I agree that the Stokes sphere is not ideal. It is a pretty hard test case (due to the spherical geometry of the falling sphere) and it will be hard to determine if your model is correct. Plus we do not have a parameter file that already uses particles for this case. On the other hand you want an example with active particles so the examples Timo suggested would also need modification.

I would start with `tests/particle_interpolator_time_dependence.prm`. It is an extremely simple model setup that still uses active particles. To make it 3D (except from setting dimension = 3) you would only need to:
- Add a Z extent to the box
- Add tangential velocity boundary conditions to the front and back boundaries
- Extend all functions to have x,y,z variables
- reduce the refinement level to keep it fast
- Add the parameter to take the bilinear least squares instead of the cell average interpolator

The model does not have an analytical solution, but you can compare your model results to the results with the cell average interpolator, and even run models at different resolution to see if they converge towards the same solution. That should be a good first test.

**Comment by egpuckett** [comment]
I agree with Rene.  Make the test that is the quickest to implement and run as possible for now, so that you finish everything that we need to finish over the summer.

Don't do a convergence test, at least not yet, just compare your model results to the results with the cell average Interpolator.

 I'm concerned that would take too much of your time away from the MLS particle interpolation development those test.

**Comment by gassmoeller** [comment]
> Don't do a convergence test, at least not yet, just compare your model results to the results with the cell average Interpolator. 

I agree, we know the method works, only the 3D case is not implemented. So if you extend it to 3D and it gives similar results as the cell average case then we can be quite sure it is correct. I also think the MLS case is more important.
